File: TP_webApp/TP_webApp/app/model/Diagram.ts
import { Layer } from "./Layer";
import { BusinessElement } from "./BusinessElement";

export class Diagram extends BusinessElement {

    protected _id: number;
    protected _layers: Layer[] = [];

<<<<<<< HEAD
    constructor(_layers: Layer[], _diagramView: DiagramView, verId: number = 0){
        super();
        this._layers = _layers ? _layers : [];
        this._verId = verId;
    }

    public initdiagramId(){
        this._diagramId = CommunicationController.instance.getNewDiagramId();
=======
    public get id(): number {
        return this._id;
>>>>>>> dev
    }

    public set id(id: number) {
        this._id = id;
    }

    public get layers(): Layer[] {
        return this._layers;
    }

    public set layers(layers: Layer[]) {
        this._layers = layers;
    }

}

File: TP_webApp/TP_webApp/Model/Diagram.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TP_webApp.Model
{
    public class Diagram
    {
        
        public Diagram(int id, List<Layer> layers)
        {
            this.id = id;
            this.layers = layers;
            
        }

        public List<Layer> layers { get; set; }
       
        public int id { get; set; }

        
    }
}
